sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/_ai_workspaces/capability_host.py

```diff
@@ -0,0 +1,226 @@
+
+    def _to_dict(self) -> Dict:
+        """Dump the object into a dictionary.
+
+        :return: Dictionary representation of the object.
+        :rtype: Dict
+        """
+
+        res: dict = CapabilityHostSchema(context={BASE_PATH_CONTEXT_KEY: "./"}).dump(self)
+        return res
+
+    @classmethod
+    def _load(
+        cls,
+        data: Optional[dict] = None,
+        yaml_path: Optional[Union[os.PathLike, str]] = None,
+        params_override: Optional[list] = None,
+        **kwargs: Any,
+    ) -> "CapabilityHost":
+        """Load a capabilityhost object from a yaml file.
+
+        :param cls: Indicates that this is a class method.
+        :type cls: class
+        :param data: Data Dictionary, defaults to None
+        :type data: Dict
+        :param yaml_path: YAML Path, defaults to None
+        :type yaml_path: Union[PathLike, str]
+        :param params_override: Fields to overwrite on top of the yaml file.
+            Format is [{"field1": "value1"}, {"field2": "value2"}], defaults to None
+        :type params_override: List[Dict]
+        :raises Exception: An exception
+        :return: Loaded CapabilityHost object.
+        :rtype: ~azure.ai.ml.entities._workspace._ai_workspaces.capability_host.CapabilityHost
+        """
+        params_override = params_override or []
+        data = data or {}
+        context = {
+            BASE_PATH_CONTEXT_KEY: Path(yaml_path).parent if yaml_path else Path("./"),
+            PARAMS_OVERRIDE_KEY: params_override,
+        }
+        return cls(
+            base_path=cast(Dict, context[BASE_PATH_CONTEXT_KEY]),
+            **load_from_dict(CapabilityHostSchema, data, context, **kwargs),
+        )
+
+    @classmethod
+    def _from_rest_object(cls, rest_obj: RestCapabilityHost) -> "CapabilityHost":
+        """Convert a REST object into a CapabilityHost object.
+
+        :param cls: Indicates that this is a class method.
+        :type cls: class
+        :param rest_obj: The REST object to convert.
+        :type rest_obj: ~azure.ai.ml._restclient.v2024_10_01_preview.models._models_py3.CapabilityHost
+        :return: CapabilityHost object.
+        :rtype: ~azure.ai.ml.entities._workspace._ai_workspaces.capability_host.CapabilityHost
+        """
+        capability_host = CapabilityHost(
+            name=str(rest_obj.name),
+            description=(rest_obj.properties.description if rest_obj.properties else None),
+            ai_services_connections=(rest_obj.properties.ai_services_connections if rest_obj.properties else []),
+            storage_connections=(rest_obj.properties.storage_connections if rest_obj.properties else []),
+            vector_store_connections=(rest_obj.properties.vector_store_connections if rest_obj.properties else []),
+            capability_host_kind=(
+                rest_obj.properties.capability_host_kind if rest_obj.properties else CapabilityHostKind.AGENTS
```

Comment:
same comment as above, do any default logic in `__init__` and not in `_from_rest_object`

Reply:
same comment like above, init do not have properties object, we are receiving it from restclient

Reply:
responded above